app/lib/publisher/backend.dart

```diff
@@ -160,6 +160,18 @@ class PublisherBackend {
       final key = _db.emptyKey.append(Publisher, id: publisherId);
       final p = (await tx.lookup<Publisher>([key])).single;
 
+      if (update.websiteUrl != null && p.websiteUrl != update.websiteUrl) {
+        final parsedUrl = Uri.tryParse(update.websiteUrl);
+        final isValid = parsedUrl != null &&
+            parsedUrl.isAbsolute &&
+            parsedUrl.toString() == update.websiteUrl;
```

Comment:
Are you sure this is a good idea? the ordering of query params might change? I haven't tested if this is an identity function for all valid URLs.

You could also test the `scheme == https` and the `host == publisherId`... and `port` not specified.

Reply:
I'll add those checks too. The equality check filters out input urls that are translated to escaped characters, and I thought that requiring immutability is not too bad at this point. We can relax it if the need arises.